src/bc/translate.rs

```diff
@@ -62,10 +62,20 @@ impl<'a, 'tcx> Program<'a, 'tcx> {
         self.krates.entry(def_id).or_insert_with(|| {
             // println!("load function {:?}", def_id);
             let cs = &context.tcx.sess.cstore;
-            let mir = cs.maybe_get_item_mir(context.tcx, def_id).unwrap_or_else(||{
-                panic!("no mir for {:?}", def_id);
-            });
-            Rc::new(context.mir_to_bytecode(&mir))
+
+            Rc::new(match context.tcx.map.as_local_node_id(def_id) {
+                Some(ref node_id) => {
+                    let mir = context.map.map.get(node_id).unwrap();
+                    context.mir_to_bytecode(mir)
+                },
+                None => {
+                    let mir = cs.maybe_get_item_mir(context.tcx, def_id).unwrap_or_else(||{
+                        panic!("no mir for {:?}", def_id);
+                    });
+                    context.mir_to_bytecode(&mir)
+                },
+            })
+
```

Comment:
Newlines at the end of a block are a no-no. If you want to visually separate things, assign the result of the `match` to a variable, and then the last line will be `Rc::new(var)`.